pkgs/servers/etcd/3_6/default.nix

```diff
@@ -0,0 +1,110 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  symlinkJoin,
+  nixosTests,
+  k3s,
+}:
+
+let
+  version = "3.6.0-rc.4";
```

Comment:
This is a pre-release version. But this will automatically update fine once the release version lands. So what we are doing here is scaffolding and testing [package & update script] for it.